### ISSUE TITLE
System.InvalidOperationException: Sequence contains no elements

### ISSUE THREAD

**Issue by nosmirck** [Tapanila/SharpCaster#89]
Hi, Thank you in advance for your support, I downloaded the sample here and when I try to run the app on Android device I get this error:

```01-30 17:43:32.901 I/MonoDroid(24977): UNHANDLED EXCEPTION:
01-30 17:43:32.906 I/MonoDroid(24977): System.InvalidOperationException: Sequence contains no elements
01-30 17:43:32.906 I/MonoDroid(24977):   at System.Linq.Enumerable.First[TSource] (System.Collections.Generic.IEnumerable`1[T] source) [0x00067] in /Users/builder/data/lanes/3511/77cb8568/source/mono/mcs/class/referencesource/System.Core/System/Linq/Enumerable.cs:956 
01-30 17:43:32.906 I/MonoDroid(24977):   at SharpCaster.Simple.Android.MainActivity+<OnCreate>d__3.MoveNext () [0x00108] in D:\R&D\ChromeCast\SharpCaster-master\SharpCaster.Simple.Android\MainActivity.cs:27 
01-30 17:43:32.906 I/MonoDroid(24977): --- End of stack trace from previous location where exception was thrown ---
01-30 17:43:32.906 I/MonoDroid(24977):   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in /Users/builder/data/lanes/3511/77cb8568/source/mono/mcs/class/referencesource/mscorlib/system/runtime/exceptionservices/exceptionservicescommon.cs:143 
01-30 17:43:32.906 I/MonoDroid(24977):   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.<ThrowAsync>m__0 (System.Object state) [0x00000] in /Users/builder/data/lanes/3511/77cb8568/source/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/AsyncMethodBuilder.cs:1018 
01-30 17:43:32.906 I/MonoDroid(24977):   at Android.App.SyncContext+<Post>c__AnonStorey0.<>m__0 () [0x00000] in /Users/builder/data/lanes/3511/ce955cc0/source/xamarin-android/src/Mono.Android/Android.App/SyncContext.cs:18 
01-30 17:43:32.906 I/MonoDroid(24977):   at Java.Lang.Thread+RunnableImplementor.Run () [0x0000b] in /Users/builder/data/lanes/3511/ce955cc0/source/xamarin-android/src/Mono.Android/Java.Lang/Thread.cs:36 
01-30 17:43:32.906 I/MonoDroid(24977):   at Java.Lang.IRunnableInvoker.n_Run (System.IntPtr jnienv, System.IntPtr native__this) [0x00009] in /Users/builder/data/lanes/3511/ce955cc0/source/monodroid/src/Mono.Android/platforms/android-24/src/generated/Java.Lang.IRunnable.cs:81 
01-30 17:43:32.906 I/MonoDroid(24977):   at (wrapper dynamic-method) System.Object:929f411e-8fc2-491e-9019-1fb85cdf3b59 (intptr,intptr)
01-30 17:43:32.927 W/art     (24977): JNI RegisterNativeMethods: attempt to register 0 native methods for android.runtime.JavaProxyThrowable
01-30 17:43:32.931 D/Mono    (24977): DllImport searching in: '__Internal' ('(null)').
01-30 17:43:32.932 D/Mono    (24977): Searching for 'java_interop_jnienv_throw'.
01-30 17:43:32.932 D/Mono    (24977): Probing 'java_interop_jnienv_throw'.
01-30 17:43:32.932 D/Mono    (24977): Found as 'java_interop_jnienv_throw'.
An unhandled exception occured.

01-30 17:43:33.950 E/mono    (24977): 
01-30 17:43:33.950 E/mono    (24977): Unhandled Exception:
01-30 17:43:33.950 E/mono    (24977): System.InvalidOperationException: Sequence contains no elements
01-30 17:43:33.950 E/mono-rt (24977): [ERROR] FATAL UNHANDLED EXCEPTION: System.InvalidOperationException: Sequence contains no elements
```

What can I do?

**Comment by Tapanila** [comment]
Hey,

Did you run the sample on emulator or a phone? That error basically means that it didn't find any chromecasts from the network

**Comment by nosmirck** [comment]
Running from the phone.

I can see 2 chromecast devices we have in the office using other apps with chromecast support or even the Android Cast Screen feature.

Is there anything missing from me? I just pulled the project, opened it in VS2015, restored nuget packages (without updating them) and run on android. As soon as it opens, after 1 or 2 seconds it crashed, I debugged and got that error message.

Also, if that's because there are no devices detected in the network, shouldn't it handle that? what about a user with no cast devices? where can I handle this exception? the error doesn't point me to any file.

Thanks in advance for any help!
